lib/terminal-command-executor.js

```diff
@@ -11,31 +11,32 @@ class TerminalCommandExecutor extends EventEmitter {
 		this.dataFinishedEventName = 'dataFinished';
 	}
 
-	run(command, destinyFolder) {
+	run(command, args, destinyFolder) {
 		this.command = command;
 		this.destinyFolder = destinyFolder;
 
-		const spawn = childProcess.spawn;
-
-		this.terminal = spawn('bash', ['-l']);
-		this.terminal.on('close', statusCode => this._streamClosed(statusCode));
-		this.terminal.stdout.on('data', data => this._stdOutDataReceived(data));
-		this.terminal.stderr.on('data', data => this._stdErrDataReceived(data));
-
-		const terminalCommand = this.destinyFolder ?
-			`cd \"${this.destinyFolder}\" && ${this.command}\n` :
-			`${this.command}\n`;
-
-		this.terminal.stdin.write(terminalCommand);
-		this.terminal.stdin.write('exit\n');
+		const folder = Array.isArray(destinyFolder) ?
+			destinyFolder[0] :
+			destinyFolder;
+
+		const process = new BufferedProcess({
```

Comment:
This is overwriting the global `process`. Use a different name like `cp` or `proc`.